tools/validate/validate.cpp

```diff
@@ -191,6 +191,18 @@ int main() {
             continue;
         }
 
+        const string relative_path = rdi->path().string();

```

Comment:
Using `native()` would be cheaper than `string()` here since it need not copy or transcode from wide to narrow character encoding. The repo is admittedly small enough that we wouldn't notice the difference, but it is growing.

Reply:
I want to upgrade this to say use `native()` because `string()` is locale sensitive. `u8string()` would be OK though.

Reply:
Now I'm using `native()` when I want direct access to `const wstring&`. When I don't have a long-lived variable (e.g. when `filepath.filename()` is the path), I'm calling `wstring()` to get a prvalue. Please meow if this is undesirable.